scipy/special/_binom.h

```diff
@@ -0,0 +1,88 @@
+// Binomial coefficient
+#pragma once
+
+#include <cmath>
+#include <limits>
+
+#include "cephes.hh"
+
+
+using std::numeric_limits;
+
+namespace scipy {
+    namespace special {
+
+	inline double binom(double n, double k) {
+	    double kx, nx, num, den, dk, sgn;
+
+	    if (n < 0) {
+		nx = std::floor(n);
+		if (n == nx) {
+		    // Undefined
+		    return numeric_limits<double>::quiet_NaN();
+		}
+	    }
+
+	    kx = std::floor(k);
+	    if (k == kx && (std::fabs(n) > 1E-8 || n == 0)) {
+		/* Integer case: use multiplication formula for less rounding
+		 * error for cases where the result is an integer.
+		 *
+		 * This cannot be used for small nonzero n due to loss of
+		 * precision. */
+		nx = std::floor(n);
+		if (nx == n && kx > nx/2 && nx > 0) {
+		    // Reduce kx by symmetry
+		    kx = nx - kx;
+		}
+
+		if (kx >= 0 && kx < 20) {
+		    num = 1.0;
+		    den = 1.0;
+		    for (int i = 1; i < 1 + static_cast<int>(kx); i++) {
+			num *= i + n - kx;
+			den *= i;
+			if (std::fabs(num) > 1E50) {
+			    num /= den;
+			    den = 1.0;
+			}
+		    }
+		    return num/den;
+		}
+	    }
+
+	    // general case
+	    if (n >= 1E10*k and k > 0) {
+		// avoid under/overflows intermediate results
+		return std::exp(-cephes::lbeta(1 + n - k, 1 + k)
+				- std::log(n + 1));
+	    }
+	    if (k > 1E8*std::fabs(n)) {
+		// avoid loss of precision
+		num = cephes::Gamma(1 + n) / std::fabs(k)
```

Comment:
Curious why is it `Gamma` with a capital 'G'?

Reply:
No clue, but cephes gamma function has been capital `G` in SciPy from the start: https://github.com/scipy/scipy/commit/52c64a94dee663b941de2b2f6592dcc6bfc4f817.